File: Loenn/entities/strawberry_jam/wormhole_booster.lua
local drawableSprite = require("structs.drawable_sprite")

local wormholeBooster = {}

wormholeBooster.name = "CommunalHelper/SJ/WormholeBooster"
wormholeBooster.depth = -11000

wormholeBooster.placements = {
    name = "booster",
    data = {
        deathColor = "",
        instantCamera = false
    }
}

wormholeBooster.fieldInformation = {
    deathColor = {
        fieldType = "color"
    }
}

local boosterColor = {120 / 255, 0, 189 / 255}
local texture = "objects/CommunalHelper/strawberryJam/boosterWormhole/boosterWormhole00"

function wormholeBooster.sprite(room, entity)
    local sprite = drawableSprite.fromTexture(texture, entity)
    sprite:setColor(boosterColor)
    return sprite
end

return wormholeBooster

File: Ahorn/entities/strawberryJam/WormholeBooster.jl
module CommunalHelperSJWormholeBooster

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/SJ/WormholeBooster" WormholeBooster(x::Integer, y::Integer, deathColor::String="61010c", instantCamera::Bool=false)


const placements = Ahorn.PlacementDict(
    "Wormhole Booster (Strawberry Jam) (Communal Helper)" => Ahorn.EntityPlacement(
        WormholeBooster,
        "rectangle"
    )
)
const boosterColor = (120,0,189, 1) ./ (255, 255, 255, 1)
function Ahorn.selection(entity::WormholeBooster)
    x, y = Ahorn.position(entity)
    sprite = "objects/CommunalHelper/strawberryJam/boosterWormhole/boosterWormhole00"

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::WormholeBooster, room::Maple.Room)
    sprite = "objects/CommunalHelper/strawberryJam/boosterWormhole/boosterWormhole00"

    Ahorn.drawSprite(ctx, sprite, 0, 0, tint=boosterColor)
end

end